src/Common/src/System/Security/Cryptography/DSASecurityTransforms.cs

```diff
@@ -0,0 +1,397 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography.Apple;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public partial class DSA : AsymmetricAlgorithm
+    {
+        public static DSA Create()
+        {
+            return new DSAImplementation.DSASecurityTransforms();
+        }
+
+        internal static partial class DSAImplementation
+        {
+            public sealed partial class DSASecurityTransforms : DSA
+            {
+                private SecKeyPair _keys;
+
+                public DSASecurityTransforms()
+                    : this(1024)
+                {
+                }
+
+                public DSASecurityTransforms(int keySize)
+                {
+                    KeySize = keySize;
+                }
+
+                public override KeySizes[] LegalKeySizes
+                {
+                    get
+                    {
+                        return new[] { new KeySizes(minSize: 512, maxSize: 1024, skipSize: 64) };
+                    }
+                }
+
+                public override int KeySize
+                {
+                    get
+                    {
+                        return base.KeySize;
+                    }
+                    set
+                    {
+                        if (KeySize == value)
+                            return;
+
+                        // Set the KeySize before freeing the key so that an invalid value doesn't throw away the key
+                        base.KeySize = value;
+
+                        _keys?.Dispose();
+                        _keys = null;
```

Comment:
Nit: slightly more code, but it's not necessary to assign null if it's already null, so could be:
```C#
if (_keys != null)
{
    _keys.Dispose();
    _keys = null;
}
```